### PR TITLE
rename metrics, change metrics to use metric classes

### PR DESCRIPTION
**Pull request by digitalsadhu**
pattern for metric names is:

`podium_<module name>_<file or class name>_<method name>`

which results in the following metric names:

* podium_client_resolver_manifest_resolve
* podium_client_resolver_fallback_resolve
* podium_client_resolver_content_resolve
* podium_client_refresh_manifests

All are histograms.

### CODE REVIEW

**Comment by trygve-lie** [comment]
Are we able to remove the [name decoration transform](https://github.com/podium-lib/client/blob/next/lib/resource.js#L21-L29)?

File: lib/resolver.content.js [hunk 1]
```diff
@@ -120,15 +120,16 @@ module.exports = class PodletClientContentResolver {
...
             };
 
-            const timer = this.metrics.timer({
-                name: 'podlet_content_request',
+            const histogram = this.metrics.histogram({
+                name: 'podium_client_resolver_content_resolve',
                 description:
                     'Time taken for success/failure of content request',
-                meta: {
+                labels: {
```
Comment (trygve-lie) [inline lib/resolver.content.js]: Ref removing the naming transform; `name` is here available on `outgoing.name`.

File: lib/resolver.content.js [hunk 2]
```diff
@@ -143,7 +144,7 @@ module.exports = class PodletClientContentResolver {
                 const resError = response.statusCode !== 200;
                 if (resError && outgoing.throwable) {
                     timer({
-                        meta: {
+                        labels: {
```
Comment (trygve-lie) [inline lib/resolver.content.js]: Ref removing the naming transform; `name` is here available on `outgoing.name`.

File: lib/resolver.content.js [hunk 3]
```diff
@@ -174,7 +175,7 @@ module.exports = class PodletClientContentResolver {
                 }
                 if (resError) {
                     timer({
-                        meta: {
+                        labels: {
```
Comment (trygve-lie) [inline lib/resolver.content.js]: Ref removing the naming transform; `name` is here available on `outgoing.name`.

File: lib/resolver.content.js [hunk 4]
```diff
@@ -214,7 +215,7 @@ module.exports = class PodletClientContentResolver {
                     contentVersion !== undefined
                 ) {
                     timer({
-                        meta: {
+                        labels: {
```
Comment (trygve-lie) [inline lib/resolver.content.js]: Ref removing the naming transform; `name` is here available on `outgoing.name`.

File: lib/resolver.content.js [hunk 5]
```diff
@@ -251,7 +255,7 @@ module.exports = class PodletClientContentResolver {
                 // Network error
                 if (outgoing.throwable) {
                     timer({
-                        meta: {
+                        labels: {
```
Comment (trygve-lie) [inline lib/resolver.content.js]: Ref removing the naming transform; `name` is here available on `outgoing.name`.

File: lib/resolver.content.js [hunk 6]
```diff
@@ -273,7 +277,7 @@ module.exports = class PodletClientContentResolver {
                 }
 
                 timer({
-                    meta: {
+                    labels: {
```
Comment (trygve-lie) [inline lib/resolver.content.js]: Ref removing the naming transform; `name` is here available on `outgoing.name`.

File: lib/resolver.content.js [hunk 7]
```diff
@@ -294,9 +298,9 @@ module.exports = class PodletClientContentResolver {
 
             r.on('end', () => {
                 timer({
-                    meta: {
+                    labels: {
```
Comment (trygve-lie) [inline lib/resolver.content.js]: Ref removing the naming transform; `name` is here available on `outgoing.name`.

File: lib/resolver.fallback.js [hunk 8]
```diff
@@ -35,15 +38,23 @@ module.exports = class PodletClientFallbackResolver {
 
     resolve(outgoing) {
         return new Promise(resolve => {
-            const timer = this.metrics.timer({
-                name: 'podlet_fallback_request',
+            const histogram = this.metrics.histogram({
+                name: 'podium_client_resolver_fallback_resolve',
                 description:
                     'Time taken for success/failure of fallback request',
+                labels: {
```
Comment (trygve-lie) [inline lib/resolver.fallback.js]: Ref removing the naming transform; `name` is here available on `outgoing.name`.

File: lib/resolver.manifest.js [hunk 9]
```diff
@@ -56,15 +59,16 @@ module.exports = class PodletClientManifestResolver {
...
 
-            const timer = this.metrics.timer({
-                name: 'podlet_manifest_request',
+            const histogram = this.metrics.histogram({
+                name: 'podium_client_resolver_manifest_resolve',
                 description:
-                    'Time taken for success/failure of mainfest request',
-                meta: {
+                    'Time taken for success/failure of manifest request',
+                labels: {
```
Comment (trygve-lie) [inline lib/resolver.manifest.js]: Ref removing the naming transform; `name` is here available on `outgoing.name`.

**Comment by trygve-lie** [comment]
To me it looks like we are able to remove the name decoration transform by setting the name where we create the metric. Each place (except in the refresh manifest method (not relevant since this one will not be bound to one podlet)) the name is available on `outgoing.name`.

**Comment by digitalsadhu** [comment]
just saw your comment, transform has now been removed :)

File: lib/resolver.content.js [hunk 10]
```diff
@@ -120,15 +120,20 @@ module.exports = class PodletClientContentResolver {
...
-                name: 'podlet_content_request',
+            const histogram = this.metrics.histogram({
+                name: 'podium_client_resolver_content_resolve',
                 description:
                     'Time taken for success/failure of content request',
-                meta: {
-                    url: reqOptions.uri,
-                    method: reqOptions.method,
+                labels: {
+                    url: outgoing.fallbackUri,
```
Comment (trygve-lie) [inline lib/resolver.content.js]: This is the resolver which fetches the content. I think using fallbackUri here is wrong. Should this be `outgoing.contentUri`?

File: lib/resolver.manifest.js [hunk 11]
```diff
@@ -56,15 +59,20 @@ module.exports = class PodletClientManifestResolver {
...
                     url: reqOptions.uri,
```
Comment (trygve-lie) [inline lib/resolver.manifest.js]: We can use `outgoing.manifestUri` instead here.

File: lib/resolver.content.js (hunk 10 above)
Comment (digitalsadhu) [reply lib/resolver.content.js]: shooot, copypasta, good catch

File: lib/resolver.manifest.js (hunk 11 above)
Comment (digitalsadhu) [reply lib/resolver.manifest.js]: came up undefined

Comment (digitalsadhu) [reply lib/resolver.manifest.js]: scratch that, works fine

**Pull request merged**